media/components/sdfv/sdfv_overrides.js

```diff
@@ -94,31 +94,121 @@ function fill_info_embedded(elem) {
             if (val === null || val === '')
                 continue;
 
-            if (attr[0] === 'debuginfo') {
-                gotoSourceBtn.on('click', function() {
-                    gotoSource(
-                        attr[1].filename,
-                        attr[1].start_line,
-                        attr[1].start_column,
-                        attr[1].end_line,
-                        attr[1].end_column
-                    );
-                });
-                gotoSourceBtn.prop('title',
-                    attr[1].filename + ':' + attr[1].start_line);
-                gotoSourceBtn.show();
-                continue;
-            }
 
-            const row = $('<tr>').appendTo(attr_table_body);
-            $('<th>', {
-                'class': 'key-col',
-                'text': attr[0],
-            }).appendTo(row);
-            $('<td>', {
-                'class': 'val-col',
-                'html': val,
-            }).appendTo(row);
+            if (attr[0] === 'instrument') {
+                if (window.instruments) {
+                    const row = $('<tr>').appendTo(attr_table_body);
+                    $('<th>', {
+                        'class': 'key-col',
+                        'text': attr[0],
+                    }).appendTo(row);
+                    const cell = $('<td>', {
+                        'class': 'val-col',
+                    }).appendTo(row);
+
+                    const select = $('<select>', {
+                        'name': 'instrument',
+                        'class': 'sdfv-property-dropdown',
+                    }).appendTo(cell);
+
+                    select.change(() => {
+                        if (elem && elem.data) {
+                            if (elem.data.attributes)
+                                elem.data.attributes.instrument = select.val();
+                            else if (elem.data.state)
+                                elem.data.state.attributes.instrument =
+                                    select.val();
+                            else if (elem.data.node)
+                                elem.data.node.attributes.instrument =
+                                    select.val();
+
+                            let g = renderer.sdfg;
+
+                            // The renderer uses a graph representation with
+                            // additional information, and to make sure that
+                            // the classical SDFG representation and that graph
+                            // representation are kept in sync, the SDFG object
+                            // is made cyclical. We use this to break the
+                            // renderer's SDFG representation back down into the
+                            // classical one, removing layout information along
+                            // with it.
```

Comment:
The function deletes properties from the renderer's sdfg object, which could be an unwanted side effect. Thus personally not a fan of using it as a global helper. It was only used here because the resulting file-change guarantees an update of SDFV, causing the renderer's graph object to be re-created.